### PR TITLE
Better errors

### PR DESCRIPTION
**Pull request by erikh**
This improves the error output of most errors in `simplebridge`. It is not a very complex patch, but is long.

### CODE REVIEW

File: extensions/simplebridge/driver_network_utils.go [hunk 1]
```diff
@@ -76,24 +75,21 @@ func (d *BridgeDriver) createBridge(id string, vlanid uint, port uint, peer, dev
 
 		linkval, err = d.getInterface(vxlan.LinkAttrs.Name, vxlan)
 		if err != nil {
-			log.Println("Error get interface", err)
-			return nil, err
+			return nil, bridgeError("retrieve interface name", err)
```
Comment (icecrime) [inline extensions/simplebridge/driver_network_utils.go:78]: `s/retrieve interface name/get interface/`?

File: extensions/simplebridge/driver_network_utils.go [hunk 2]
```diff
@@ -13,12 +12,15 @@ import (
 	"github.com/vishvananda/netlink/nl"
 )
 
+func bridgeError(typ string, err error) error {
```
Comment (icecrime) [inline extensions/simplebridge/driver_network_utils.go:15]: That's really a matter of taste, but I'd probably nest it inside the calling function if there's a single one (applies to other `*Error()` functions in the patch).

**Comment by icecrime** [comment]
Yay, debugging this branch was a pain :+1:

File: extensions/simplebridge/driver_network_utils.go (hunk 2 above)
Comment (erikh) [inline extensions/simplebridge/driver_network_utils.go:15]: you mean as an unnamed func? I guess I could do that, but I don't really see the value in it. Can you explain what the advantage is over a top-level function?

Comment (icecrime) [inline extensions/simplebridge/driver_network_utils.go:15]: Not much besides making it obvious that it's not used anywhere else. But don't bother, I'll merge like this.

**Pull request merged**